ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -481,14 +481,20 @@ void groupRemove(boolean deleteDirectory, boolean renameDirectory) {
     /* Shutdown is triggered here inorder to avoid any locked files. */
     state.getStateMachineUpdater().setRemoving();
     close();
+    if (state.getStateMachineUpdater().getError() != null) {
+      LOG.error("{}: groupRemove failed: {}. Falling back to directory rename.",
+              getGroup().getGroupId(), getMemberId(), state.getStateMachineUpdater().getError());
+      deleteDirectory = false;
+      renameDirectory = true;
```

Comment:
It seems wrong to change the request silently.  We need some design on this.

Reply:
done